### DISCUSSION TITLE
Jamstack survey 2021

### DISCUSSION THREAD

**Discussion by tomap** [hexojs/hexo#4801] [Ideas]
https://jamstack.org/survey/2021/

![Screenshot_20211013-211128](https://user-images.githubusercontent.com/153854/137198277-5b3d1d66-f349-45e5-957d-359c53bcc7dd.png)

Not good for hexo!

I feel this is just a perception issue. We can do better!

A few blog posts about how to solve trendy issues like data driven site, use tailwind, deploy on cloud flare pages, ... And we could be back in the game ✊

**Comment by SukkaW** [comment]
Hexo is born in the dark age of web development when there is no Vue, no React, no Angular, no Virtual DOM, no MVVM design model. 

Nowadays, those popular static site generators and frameworks are based on Vue or React, while Hexo is still highly relied on old-school template languages like `ejs` and `pug`.

We need maintainers to migrate Hexo to the monorepo, which will be helpful for us to manage/publish so many packages.
We need maintainers to migrate Hexo to the typescript, which enables easier development for other contributors.
We need maintainers to completely rewrite our documentation. E.g. Hexo can be deployed to Cloudflare Pages, but nobody there trying to add documentation for that.

In short, we need maintainers, a LOT of maintainers.

----

Also here are some of my personal ideas:

- Hexo supports users to write their posts in any syntax they want as long as there is corresponded renderer plugin. But most of the Hexo users only use Markdown. So we should drop support for other syntax and supports markdown only. E.g. we should only allow writing posts in Markdown, make things a lot easier.
  - We don't need to make a highlight plugin for every template (so no `backtick_code_filter`), only for markdown.
  - We don't need to let the ToC plugin parses HTML: markdown is much easier to work with.
- Nunjucks is great. But it is also outdated and very hard when in the markdown. We should introduce a new Tag Plugin Syntax that does not rely on Nunjucks. VuePress supports JSX in the markdown, we should try that.
  - We should use `markdown-it` as it supports JSX in Markdown while `marked.js` doesn't.
  - We should make a migrator for users to migrate their old Nunjucks syntax to JSX syntax.
- We need to make some plugins built-in by default: like RSS and search.
- We should support modern bundler features (not those gulp's approach), at least make it easier for theme developers and site owners to work with Tailwind, PostCSS, TypeScript, Babel, etc, etc.

**Reply by yoshinorin** [reply]
I disagree with `built-in by default: like RSS and search`. I like to separate them like UNIX philosophy. ("Do One Thing and Do It Well") IMHO, we should separate some build-in features. (e.g `jsfiddle`, `youtube` etc...) After we migrate to monorepo I would like to separate them.

But, other ideas totally agree :)

**Comment by tomap** [comment]
About switching to markdown-it https://github.com/hexojs/hexo-starter/pull/52

**Reply by SukkaW** [reply]
I am also considering https://mdxjs.com/ which is used by Gatsby.

**Comment by tomap** [comment]
Also, drop Git from pre requisites. Only used for the hexo init command. Could be embedded into hexo-cli or downloaded via an npm package or GitHub releases

**Reply by tomap** [reply]
https://stackoverflow.com/a/20645423/383029 seems simple!
